ring_buffer/ring_buffer.py

```diff
@@ -8,25 +8,82 @@ def __init__(self, capacity):
         self.storage = DoublyLinkedList()
 
     def append(self, item):
-        pass
+        if self.storage.length == 0:
+            self.storage.add_to_head(item)
+            self.current = self.storage.head
+            #nothing in DLL, add item to start
+        elif self.storage.length < self.capacity:
+            self.storage.add_to_tail(item)
+            self.current = self.storage.tail
+            #still have room in DLL, add item to list
+        elif self.current is self.storage.tail:
```

Comment:
so, if it fails the above condition and is still the tail... I see, remove the head node, add to the head, and reset the system. DLLs definitely make this much easier than if it was a SLL hahaha.